### PR TITLE
feat(j-s): Display indictment cases for FMST

### PR DESCRIPTION
**Pull request by unakb**
[Mál til fullnustu](https://app.asana.com/0/1199153462262248/1207612392953280)

## What

New table for FMST to view indictment cases

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Added `indictmentCompletedDate` to case details.
  - Introduced enhanced logic for calculating indictment verdict appeal deadlines.
  - Implemented new message definitions for court and sentencing dates.
  - Added comprehensive filtering and categorization for indictment cases.
  - Updated user interface to display indictment cases distinctively for prison administration roles.

- **Bug Fixes**
  - Corrected typo from "indicitment" to "indictment" in test descriptions.

- **Tests**
  - Added tests for `getIndictmentVerdictAppealDeadline` function to ensure accuracy.

- **Refactor**
  - Improved case filtering logic and updated type declarations to enhance code clarity and maintainability.

### CODE REVIEW

File: apps/judicial-system/api/src/app/modules/case/interceptors/case.transformer.ts [hunk 1]
```diff
@@ -117,22 +122,18 @@ export const getIndictmentInfo = (
...
-      indictmentInfo.indictmentVerdictAppealDeadline = expiryDate.toISOString()
-    }
+    const verdictAppealDeadline =
+      getIndictmentVerdictAppealDeadline(verdictViewDates)
+    indictmentInfo.indictmentVerdictAppealDeadline = verdictAppealDeadline
+      ? verdictAppealDeadline.toISOString()
+      : undefined
+
+    indictmentInfo.indictmentVerdictViewedByAll =
+      indictmentInfo.indictmentVerdictAppealDeadline ? true : false
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/api/src/app/modules/case/interceptors/case.transformer.ts:132]: I prefer the first way

**Review by oddsson — APPROVED** [review]
Nice!

File: apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql [hunk 2]
```diff
@@ -94,6 +94,7 @@ query Cases {
     indictmentVerdictAppealDeadline
     indictmentRulingDecision
     indictmentDecision
+    indictmentCompletedDate
```
Comment (oddsson) [inline apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql:97]: I think this is causing an error in schema generation. Is this here by accident or did you mean to add this to the case query as well?

Comment (unakb) [reply apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql:97]: Yeees true actually, I did mean to add it in there but I forgot to add it in to the model in the API layer, fixing now

**Pull request merged**